### PR TITLE
Implement some serialization better

### PR DESCRIPTION
**Pull request by AlexProgrammerDE**

### CODE REVIEW

File: protocol/src/main/java/org/geysermc/mcprotocollib/protocol/packet/ingame/serverbound/ServerboundChatPacket.java [hunk 1]
```diff
@@ -26,12 +26,11 @@ public ServerboundChatPacket(ByteBuf in, MinecraftCodecHelper helper) {
         this.message = helper.readString(in);
         this.timeStamp = in.readLong();
         this.salt = in.readLong();
-        if (in.readBoolean()) {
-            this.signature = new byte[256];
-            in.readBytes(this.signature);
-        } else {
-            this.signature = null;
-        }
+        this.signature = helper.readNullable(in, inBuf -> {
```
Comment (basaigh) [inline protocol/src/main/java/org/geysermc/mcprotocollib/protocol/packet/ingame/serverbound/ServerboundChatPacket.java]: inBuf -> buf to be consistent.

**Review by basaigh — COMMENTED** [review]
Other than this, lgtm.

File: protocol/src/main/java/org/geysermc/mcprotocollib/protocol/packet/ingame/serverbound/ServerboundChatPacket.java (hunk 1 above)
Comment (AlexProgrammerDE) [reply protocol/src/main/java/org/geysermc/mcprotocollib/protocol/packet/ingame/serverbound/ServerboundChatPacket.java]: I really hate having to name them. It's all literally writing to the same buf.

**Review by Redned235 — APPROVED** [review]

**Review by basaigh — APPROVED** [review]

**Review by onebeastchris — APPROVED** [review]
lgtm

**Pull request merged**